modules/apps/frontend-js/frontend-js-walkthrough-web/src/main/resources/META-INF/resources/Walkthrough.js

```diff
@@ -126,86 +94,68 @@ const ALIGNMENTS_GUESS_MAP = {
 	trbr: 'top',
 };
 
-const WalkthroughOverlay = ({popoverVisible, trigger}) => {
-	const [overlayBounds, setOverlayBounds] = useState({
-		height: 0,
-		width: 0,
-		x: 0,
-		y: 0,
-	});
-
-	const updateOverlayBounds = useCallback(
-		(overlayBounds) => {
-			overlayBounds = overlayBounds
-				? overlayBounds
-				: trigger.getBoundingClientRect();
-
-			setOverlayBounds({
-				height: overlayBounds.height,
-				width: overlayBounds.width,
-				x: overlayBounds.x,
-				y: overlayBounds.y,
-			});
-		},
-		[trigger]
-	);
-
-	useEffect(() => {
-		updateOverlayBounds();
-	}, [updateOverlayBounds]);
-
-	useObserveRect(updateOverlayBounds, trigger);
-
-	return (
-		<OverlayMask
-			bounds={overlayBounds}
-			onBoundsChange={setOverlayBounds}
-			visible={popoverVisible}
-		/>
-	);
-};
-
-const Walkthrough = ({closeOnClickOutside, closeable, skippable, steps}) => {
-	const [currentStepIndex, setCurrentStepIndex] = useState(0);
-
+const Step = ({
+	closeOnClickOutside,
+	closeable,
+	currentStep,
+	onCurrentStep,
+	pages,
+	skippable,
+	steps,
+}) => {
 	const popoverRef = useRef(null);
 
 	const hotspotRef = useRef(null);
 
-	const [popoverVisible, setPopoverVisible] = useState(false);
+	const [popoverVisible, setPopoverVisible] = useLocalStorage(
+		'walkthrough-popover-visible',
+		false
+	);
 
 	const {
 		content,
 		darkbg,
+		id,
+		next,
 		positioning: defaultPositioning = 'right-top',
+		previous,
 		title,
-	} = steps[currentStepIndex];
+	} = steps[currentStep];
+
+	const path = location.pathname + location.search + location.hash;
 
 	const [currentAlignment, setCurrentAlignment] = useState(
 		defaultPositioning
 	);
 
 	const memoizedTrigger = useMemo(() => {
-		const currentNode = steps[currentStepIndex].nodeToHighlight;
+		const currentNode = steps[currentStep].nodeToHighlight;
 
 		if (currentNode) {
 			return document.querySelector(currentNode);
 		}
-	}, [steps, currentStepIndex]);
+	}, [steps, currentStep]);
 
 	const previousTrigger = usePrevious(memoizedTrigger);
 
 	const changeStep = useCallback(
-		(isNext, index) => {
+		(action, isNext, index) => {
 			const hasElement = document.querySelector(
 				steps[index].nodeToHighlight
 			);
 
-			if (hasElement) {
-				setCurrentStepIndex(index);
+			if (
+				hasElement ||
+				(action && pages[action].includes(steps[index].id))
+			) {
+				onCurrentStep(index);
 			}
 			else {
-				setCurrentStepIndex(
+				console.error(
```

Comment:
That was the decision in the ticket https://issues.liferay.com/browse/LPS-154701, ideally, we don't want to show anything to the user because that would be a bug and not interesting for the user instead of this we skip this step and we move to the next one if it exists otherwise keep the current step.